### PR TITLE
Fix collision and performance issue

### LINKED ISSUE #210: Collision and performance issue
Hello,

First and foremost, a big thank you for your dedication to maintaining this library. We genuinely appreciate your hard work.

We've been using this library in our production-grade application and came across a couple of issues that we'd like to discuss and propose solutions for.

**Background**:
1. **Collision**:
   The `url_key` column is currently case-insensitive, which has led to it treating two different keys like `aLlen` and `allen` as identical. To address this, we believe it would be beneficial to resolve this at the package level. This way, users won't need to be concerned about case sensitivity when utilizing the library.

2. **Performance**:
   The library currently fetches the latest ID using `ShortUrl::latest()`, which, while functional, is not the most performant approach. It sorts the rows by `created_at` date, which can be resource-intensive. A more efficient method would be to use `ShortUrl::max('id')`, as the `id` column is indexed by default. This change could significantly boost performance for users.

We'd love to hear your thoughts on these proposed changes. If you're open to it, we can prepare a pull request to implement these improvements.

Alternatively, for the **collision** issue, we could consider adding a note to the documentation section, explaining why users might encounter this situation and how to handle it.

Once again, thank you sincerely for your time and effort in maintaining this package.

Best regards,
Hari Lamichhane

### PR DESCRIPTION
**Pull request by hari**
Closes #210

### CODE REVIEW

**Comment by hari** [comment]
@ash-jc-allen This one needs your review 😊. All tests are passing. I don't think I need to add any extra tests.

**Comment by ash-jc-allen** [comment]
Hey @hari! Apologies for the delay in getting this merged (it's been a busy few weeks!). Huge thank you 😄

**Comment by boris-glumpler** [comment]
Hey @ash-jc-allen, did you see my comment above? Came across this same issue with another package a while ago and their migrations threw an error for me because this particular collation doesn't exist on Postgres, which is case-sensitive by default.

**Comment by ash-jc-allen** [comment]
Hey @boris-glumpler! I can't see any comments from you on this page, sorry. This is all I see:

image

I wonder if the comment might be part of a "review" that's not been submitted yet, maybe?

Thanks for pointing this out to me. I've not got any experience with using Postgres, so I wasn't aware of that. I'll make sure to update the readme to mention that you might want to remove this part of the migration if you're using Postgres 🙂

**Comment by boris-glumpler** [comment]
Duh... You  should be able to see it now ;)

File: database/migrations/2019_12_22_015115_create_short_urls_table.php [hunk 1]
```diff
@@ -16,7 +16,7 @@ public function up()
         Schema::create('short_urls', function (Blueprint $table) {
             $table->bigIncrements('id');
             $table->text('destination_url');
-            $table->string('url_key')->unique();
+            $table->string('url_key')->unique()->collation('utf8mb4_bin');
```
[the remark this answers is not in the document]
Comment (ash-jc-allen) [reply database/migrations/2019_12_22_015115_create_short_urls_table.php:19]: Nice one, thanks for this suggestion! I can't quite make my mind up between whether we should automatically handle this in the migration, or just mention it in the docs so people can make the change themselves. I guess there are pros and cons to either side 🤔

What's your opinion on this? 🙂

[the remark this answers is not in the document]
Comment (hari) [reply database/migrations/2019_12_22_015115_create_short_urls_table.php:19]: @boris-glumpler thank you for the information. I think we should do both - update the doc and add the conditional logic @ash-jc-allen

[the remark this answers is not in the document]
Comment (boris-glumpler) [reply database/migrations/2019_12_22_015115_create_short_urls_table.php:19]: Not sure there's a downside to adding the conditional. If it bothers people, they can change it. If not, then no harm done.

**Pull request merged**